server/weaverbird/backends/sql_translator/steps/evolution.py

```diff
@@ -0,0 +1,73 @@
+from weaverbird.backends.sql_translator.steps.utils.query_transformation import (
+    build_selection_query,
+)
+from weaverbird.backends.sql_translator.types import (
+    SQLPipelineTranslator,
+    SQLQuery,
+    SQLQueryDescriber,
+    SQLQueryExecutor,
+    SQLQueryRetriever,
+)
+from weaverbird.pipeline.steps import EvolutionStep
+
+
+def translate_evolution(
+    step: EvolutionStep,
+    query: SQLQuery,
+    index: int,
+    sql_query_retriever: SQLQueryRetriever = None,
+    sql_query_describer: SQLQueryDescriber = None,
+    sql_query_executor: SQLQueryExecutor = None,
+    sql_translate_pipeline: SQLPipelineTranslator = None,
+    subcall_from_other_pipeline_count: int = None,
+) -> SQLQuery:
+    """
+    Example of generated query (evolution on date column + 1 year, groupby orderstatus):
+    SELECT A.O_TOTALPRICE AS P1, B.O_TOTALPRICE AS P2, A.O_ORDERDATE AS DATE, A.O_ORDERSTATUS as STAT,
+    B.O_ORDERSTATUS AS STATB
+    FROM ORDERS_TRUNC A
+    LEFT JOIN ORDERS_TRUNC B
+    ON A.O_ORDERDATE = DATEADD('year', 1, B.O_ORDERDATE) AND A.O_ORDERSTATUS = B.O_ORDERSTATUS
+    WHERE DATEADD('year', 1, B.O_ORDERDATE) IS NULL OR B.O_ORDERSTATUS IS NULL
+    """
+
+    DATE_UNIT = {
+        'vsLastYear': 'year',
+        'vsLastMonth': 'month',
+        'vsLastWeek': 'week',
+        'vsLastDay': 'day',
+    }
+
+    query_name = f'EVOLUTION_STEP_{index}'
+    new_column_name = (
+        step.new_column.upper()
+        if step.new_column
+        else f'{step.value_col}_EVOL_{step.evolution_format.upper()}'
+    )
+
+    if step.evolution_format == 'abs':
+        new_column = f'''A.{step.value_col} - B.{step.value_col} AS {new_column_name}'''
+    else:
+        new_column = f'''A.{step.value_col} / B.{step.value_col} - 1 AS {new_column_name}'''
```

Comment:
I would have suggested with some parentheses : 
```python
    if step.evolution_format == 'abs':
        new_column = f'''(A.{step.value_col} - B.{step.value_col}) AS {new_column_name}'''
    else:
        new_column = f'''((A.{step.value_col} / B.{step.value_col}) - 1) AS {new_column_name}'''
```

Not sure yet if that will break something but it could be nice to scope operations like this.... i don't know 🤔

Reply:
:+1: